src/openrct2/rct2/S6Importer.cpp

```diff
@@ -1372,31 +1372,37 @@ class S6Importer final : public IParkImporter
         dst->colours = src->colours;
         dst->track_progress = src->track_progress;
         dst->track_direction = src->track_direction;
+        dst->TrackLocation = { src->track_x, src->track_y, src->track_z };
         if (src->boat_location.isNull() || static_cast<RideMode>(ride.mode) != RideMode::BoatHire
             || src->status != static_cast<uint8_t>(Vehicle::Status::TravellingBoat))
         {
             dst->BoatLocation.setNull();
-            dst->SetTrackType(src->GetTrackType());
-            dst->SetTrackDirection(src->GetTrackDirection());
+            // RotationControlToggle and Booster are saved as the same track piece ID
+            // Which one the vehicle is using must be determined
+            if (src->GetTrackType() == TrackElemType::RotationControlToggleAlias)
+            {
+                // Merging hacks mean the track type that's appropriate for the ride type is not necessarily the track type the
+                // ride is on. It's possible to create unwanted behavior if a user layers spinning control track on top of
+                // booster track but this is unlikely since only two rides have spinning control track - by default they load as
+                // booster
+                TileElement* tileElement2 = map_get_track_element_at_of_type_seq(
+                    dst->TrackLocation, TrackElemType::RotationControlToggle, 0);
+
+                if (tileElement2 != nullptr)
```

Comment:
You're not accounting for a bunch of conditions now. You want it to fall back to a normal Set(Get()) if this fails. Also you are no longer setting the direction

Reply:
I'm not sure what you mean by "a bunch of conditions." There's only 3 possible cases:
- vehicle's boat location is set
- vehicle is on RotationControlToggleAlias
- vehicle is not on RotationControlToggleAlias

Regardless, the assignment at 1374 should be changed to the getter/setter